### PR TITLE
Non-ambiguous ORDER BY col should not error

### PR DESCRIPTION
**Pull request by max-hoffman**

### CODE REVIEW

**Review by zachmu — APPROVED** [review]

File: sql/planbuilder/parse_test.go [hunk 1]
```diff
@@ -41,10 +41,23 @@ type planTest struct {
 
 func TestPlanBuilder(t *testing.T) {
 	var verbose, rewrite bool
-	//verbose = true
-	//rewrite = true
+	verbose = true
```
Comment (nicktobey) [inline sql/planbuilder/parse_test.go:44]: Revert these.

**Review by nicktobey — APPROVED** [review]

File: sql/planbuilder/parse_test.go [hunk 2]
```diff
@@ -45,6 +45,19 @@ func TestPlanBuilder(t *testing.T) {
 	//rewrite = true
 
 	var tests = []planTest{
+		{
```
Comment (JCOR11599) [inline sql/planbuilder/parse_test.go:48]: This apparently works in MySQL
```
select t1.x as x, t1.x as x from xy t1, xy t2 order by x;
```


but this fails in MySQL
```
select t1.x as x, t2.x as x from xy t1, xy t2 order by x;
```

I think they both fail in dolt even with your fix.

Also should make sure
```
select x, y as x from xy order by x;
```
still fails

Comment (max-hoffman) [reply sql/planbuilder/parse_test.go:54]: good catch, take another peek see if the solution's OK

Comment (nicktobey) [reply sql/planbuilder/parse_test.go:54]: I previously outlined the exact resolution rules for `ORDER BY` in MySQL here: https://github.com/dolthub/dolt/issues/6676

**Review by JCOR11599 — APPROVED** [review]
LGTM

**Pull request merged**